File: runtime/java/src/main/java/dev/argon/verilization/java_runtime/FormatWriter.java
package dev.argon.verilization.java_runtime;

import java.io.IOException;

public interface FormatWriter {
	void writeByte(byte b) throws IOException;
	void writeShort(short s) throws IOException;
	void writeInt(int i) throws IOException;
	void writeLong(long l) throws IOException;
	void writeBytes(byte[] data) throws IOException;
}

File: runtime/scala/src/main/scala/dev/argon/verilization/scala_runtime/FormatWriter.scala
package dev.argon.verilization.scala_runtime

import zio.{ZIO, Chunk}
import scala.{Int => SInt}

trait FormatWriter[-R, +E] {
    def writeByte(b: Byte): ZIO[R, E, Unit]
    def writeShort(s: Short): ZIO[R, E, Unit]
    def writeInt(i: SInt): ZIO[R, E, Unit]
    def writeLong(l: Long): ZIO[R, E, Unit]
    def writeBytes(data: Chunk[Byte]): ZIO[R, E, Unit]
}
